applications/MPMApplication/custom_conditions/particle_based_conditions/mpm_particle_base_dirichlet_condition.cpp

```diff
@@ -66,6 +66,74 @@ void MPMParticleBaseDirichletCondition::FinalizeSolutionStep( const ProcessInfo&
     m_imposed_displacement = ZeroVector(3);
 }
 
+void MPMParticleBaseDirichletCondition::CalculateInterfaceContactForce(const ProcessInfo& rCurrentProcessInfo )
+{
+    const unsigned int number_of_nodes = GetGeometry().PointsNumber();
+
+    GeneralVariables Variables;
+    MPMShapeFunctionPointValues(Variables.N);
+
+    array_1d<double, 3 > mpc_force = ZeroVector(3);
+
+    for ( unsigned int i = 0; i < number_of_nodes; i++ )
+    {
+        if (Variables.N[i] > std::numeric_limits<double>::epsilon() )
+        {
+            auto r_geometry = GetGeometry();
+            
+            const double nodal_area  = r_geometry[i].FastGetSolutionStepValue(NODAL_AREA, 0);
+            const Vector nodal_force = r_geometry[i].FastGetSolutionStepValue(REACTION);
+
+            if (nodal_area > std::numeric_limits<double>::epsilon())
+            {
+                mpc_force += Variables.N[i] * nodal_force * this->GetIntegrationWeight() / nodal_area;
+            }
+        }
+    }
+
+    // Apply in the normal contact direction and allow releasing motion
+    if (Is(CONTACT))
+    {
+        // Apply only in the normal direction
+        const double normal_force = MathUtils<double>::Dot(mpc_force, m_normal);
+
+        // This check is done to avoid sticking forces
+        if (normal_force > 0.0)
+            mpc_force = 1.0 * normal_force * m_normal;
```

Comment:
```suggestion
            mpc_force = normal_force * m_normal;
```

Reply:
`mpc_force` is overwritten here, meaning that lines 78-92 are unnecessary if `Is(CONTACT)` is `true`.

Reply:
it's true that it is overwritten but before it is used to calculate normal_force. Therefore the lines above are needed in any case.

Reply:
Got it.